src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorHelpers.cs

```diff
@@ -0,0 +1,641 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Uno.UI.Helpers.WinUI;
+using Windows.Foundation;
+using Windows.System.Threading;
+using Windows.UI;
+using Windows.UI.Core;
+using Windows.UI.Xaml.Media.Imaging;
+
+namespace Microsoft.UI.Xaml.Controls.Primitives
+{
+	internal static class ColorHelpers
+	{
+		public const int CheckerSize = 4;
+
+		// Uno Doc: These enums were originally in ColorHelpers.h in WinUI.
+		// They could be moved to the namespace instead of the class here
+		public enum IncrementDirection
+		{
+			Lower,
+			Higher,
+		};
+
+		public enum IncrementAmount
+		{
+			Small,
+			Large,
+		};
+
+		public static Hsv IncrementColorChannel(
+			Hsv originalHsv,
+			ColorPickerHsvChannel channel,
+			IncrementDirection direction,
+			IncrementAmount amount,
+			bool shouldWrap,
+			double minBound,
+			double maxBound)
+		{
+			Hsv newHsv = originalHsv;
+
+			if (amount == IncrementAmount.Small || !DownlevelHelper.ToDisplayNameExists())
+			{
+				// In order to avoid working with small values that can incur rounding issues,
+				// we'll multiple saturation and value by 100 to put them in the range of 0-100 instead of 0-1.
+				newHsv.S *= 100;
+				newHsv.V *= 100;
+
+				// Uno Doc: *valueToIncrement removed for C#
+				double valueToIncrement = 0.0;
+				double incrementAmount = 0.0;
+
+				// If we're adding a small increment, then we'll just add or subtract 1.
+				// If we're adding a large increment, then we want to snap to the next
+				// or previous major value - for hue, this is every increment of 30;
+				// for saturation and value, this is every increment of 10.
+				switch (channel)
+				{
+					case ColorPickerHsvChannel.Hue:
+						valueToIncrement = newHsv.H;
+						incrementAmount = amount == IncrementAmount.Small ? 1 : 30;
+						break;
+
+					case ColorPickerHsvChannel.Saturation:
+						valueToIncrement = newHsv.S;
+						incrementAmount = amount == IncrementAmount.Small ? 1 : 10;
+						break;
+
+					case ColorPickerHsvChannel.Value:
+						valueToIncrement = newHsv.V;
+						incrementAmount = amount == IncrementAmount.Small ? 1 : 10;
+						break;
+
+					default:
+						throw new Exception("Invalid ColorPickerHsvChannel."); // Uno Doc: 'winrt::hresult_error(E_FAIL);'
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)